### ISSUE TITLE
Forge site directory not deleted in teardown

### ISSUE THREAD

**Issue by dev-anthonyrae** [mehrancodes/laravel-harbor#114]
When the teardown command runs, the forge site is removed, but the actual folder on the server still exists. 
Is this the expected behaviour? 
Is it possible to automate the deletion of the the site directory on the server?

**Comment by mehrancodes** [comment]
@dev-anthonyrae The directory should get removed by the Forge in 10 seconds after the teardown commend finishes. There might be nothing Harbor can do about this.

**Comment by keybrdist** [comment]
@mehrancodes - We are having the same issue, so I did some testing.

1. The issue is only happening when Harbor issues the teardown
2. I tested directly with the Forge API (no issue found)
3. I tested using the Forge SDK (no issue found)

Site files are successfully deleted using SDK and API but not when issued via Harbor.

**Comment by mehrancodes** [comment]
@keybrdist, @dev-anthonyrae I just tested it, and it removes the folder on the server. Maybe I can try it again with your Harbor .env public keys?

**Comment by keybrdist** [comment]
@mehrancodes I'll try running harbor locally to do some testing.  Is it just a matter of setting the env vars and then running `./vendor/bin/harbor teardown` ?

**Comment by mehrancodes** [comment]
> @mehrancodes I'll try running harbor locally to do some testing.  Is it just a matter of setting the env vars and then running `./vendor/bin/harbor teardown` ?

@keybrdist, You can do that by running **php harbor teardown**. Same as when you want to provision a site

It’s built on Laravel Zero, but used for console apps. Let me know if you need any more help!

**Comment by keybrdist** [comment]
I have run some tests.. 

tldr; I have to run `harbor teardown` twice to delete the site directory.  The first time I run `harbor teardown`, the site and database is deleted and disappears from the forge UI; however the directory still remains.  When I run it the 2nd time, the site directory is finally deleted.

---

I setup  a base projects with the following:
`composer.json`
```
{
    "require": {
        "mehrancodes/laravel-harbor": "^1.0"
    }
}
```


A basic `teardown.sh` script:
```
#!/bin/bash

set -e
export FORGE_GIT_BRANCH=xxxxxxx
export FORGE_SERVER=1234
export FORGE_DB_NAME=pr_1128
export SUBDOMAIN_NAME=pr-1128
export FORGE_DOMAIN=tst.xxxxxx.com
export FORGE_GIT_REPOSITORY=xxxxx/xxxxx
export FORGE_TOKEN=eyJ0.....

./vendor/bin/harbor teardown
```

Output:
```
➜ ./teardown.sh                    
  INFO    Start finding the server.
  INFO    Finding the associated site.
  INFO    Removing scheduled command.
  INFO    Removing database with user.
  SUCCESS    Environment teardown successful! All provisioned resources have been removed.
(base) 

➜ ./teardown.sh                    
  INFO    Start finding the server.
  INFO    Finding the associated site.
  SUCCESS    Environment teardown successful! All provisioned resources have been removed.
(base) 

➜ ./teardown.sh
  INFO    Start finding the server.
  INFO    Finding the associated site.
  FAIL    ---> Site not found.
(base) 
```

^^ notice on the 3rd attempt it finally says "Site not found".  I had to run it twice to remove all resources.

**Comment by keybrdist** [comment]
UPDATE: I added sleep(30); to DestroySite.php and it successfully deleted the directory as well.

I suspect the delete operation is being called before the other operations are actually complete, and that the Forge API operations are asyncronous and have not completely before DestroySite is called.

```php
class DestroySite
{
    use Outputifier;

    public function __invoke(ForgeService $service, Closure $next)
    {
        sleep(30);
        $service->site->delete();

        return $next($service);
    }
}
```

Note: I think the above is a good test but perhaps it would be better if you add something more dynamic to make sure previous operations in the pipelines have been been completed before DestroySite is invoked.

**Comment by keybrdist** [comment]
> @keybrdist, @dev-anthonyrae I just tested it, and it removes the folder on the server. Maybe I can try it again with your Harbor .env public keys?

btw, regarding your testing -- This might only be affecting deployments that have other resources. For example.. all my failed tests were launched with FORGE_JOB_SCHEDULER=true. I'm guessing that `DestroySite` is being called before `RemoveTaskScheduler` is completed. hth

**Comment by keybrdist** [comment]
@mehrancodes sorry to tag you. Just wanted to make sure you saw my latest comments on this matter.

**Comment by olivernybroe** [comment]
Hello 👋 

When deleting a site in Forge, Forge will automatically delete all related jobs and daemons. 
So the `RemoveInertiaSupport` and `RemoveTaskScheduler` could be removed here as Forge will automatically do this. 

It will require however that enabling inertia is done via the endpoint for enabling Inertia, so Forge knows that the daemon is related to that site. 
https://forge.laravel.com/api-documentation#enablecreate-inertia-daemon

The same is also true for the deploy key, when deleting a site, Forge will delete the deploy key from Forge also (not the source control providers).

**Comment by mehrancodes** [comment]
@olivernybroe, @keybrdist, Thanks for following up! Everything's working great without the `RemoveTaskScheduler`. I've opened a PR to remove it since we don't need it anymore: https://github.com/mehrancodes/laravel-harbor/pull/120.

Let's see if it works smoothly now without `RemoveTaskScheduler` :)

Feel free to open a PR for `RemoveInertiaSupport` if you'd like to tackle it this week.

**Comment by olivernybroe** [comment]
@mehrancodes thanks for fixing it, we will report back to the user that reported having this issue :) 

Forgot to put disclaimer that I work for Forge 😅

**Comment by keybrdist** [comment]
@olivernybroe I think that user is me! 🙏
